core/overdrive.py

```diff
@@ -468,7 +447,8 @@ def token_authorization_header(self) -> str:
     def fulfillment_authorization_header(self) -> str:
         is_test_mode = (
             True
-            if self._server_nickname == OverdriveConfiguration.TESTING_SERVERS
```

Comment:
This change won't work as expected, since `self._server_nickname` accounts for production servers being the default, whereas `self._configuration.server_nickname` does not.